packages/integrations/src/debug.ts

```diff
@@ -1,77 +1,59 @@
-import type { Client, Event, EventHint, EventProcessor, Hub, Integration } from '@sentry/types';
+import { convertIntegrationFnToClass } from '@sentry/core';
+import type { Event, EventHint, IntegrationFn } from '@sentry/types';
 import { consoleSandbox } from '@sentry/utils';
 
+const INTEGRATION_NAME = 'Debug';
+
 interface DebugOptions {
   /** Controls whether console output created by this integration should be stringified. Default: `false` */
   stringify?: boolean;
   /** Controls whether a debugger should be launched before an event is sent. Default: `false` */
   debugger?: boolean;
 }
 
-/**
- * Integration to debug sent Sentry events.
- * This integration should not be used in production
- */
-export class Debug implements Integration {
-  /**
-   * @inheritDoc
-   */
-  public static id: string = 'Debug';
-
-  /**
-   * @inheritDoc
-   */
-  public name: string;
-
-  private readonly _options: DebugOptions;
-
-  public constructor(options?: DebugOptions) {
-    this.name = Debug.id;
-
-    this._options = {
-      debugger: false,
-      stringify: false,
-      ...options,
-    };
-  }
-
-  /**
-   * @inheritDoc
-   */
-  public setupOnce(
-    _addGlobalEventProcessor: (eventProcessor: EventProcessor) => void,
-    _getCurrentHub: () => Hub,
-  ): void {
-    // noop
-  }
-
-  /** @inheritdoc */
-  public setup(client: Client): void {
-    if (!client.on) {
-      return;
-    }
-
-    client.on('beforeSendEvent', (event: Event, hint?: EventHint) => {
-      if (this._options.debugger) {
-        // eslint-disable-next-line no-debugger
-        debugger;
+const debugIntegration = ((options: DebugOptions = {}) => {
+  const _options = {
+    debugger: false,
+    stringify: false,
+    ...options,
+  };
+
+  return {
+    name: INTEGRATION_NAME,
+    setup(client) {
+      if (!client.on) {
```

Comment:
l: let's add todo v8 comment to remove this?

Reply:
We have a whole bunch of these checks already anyhow, I think we'll do a general search for this when we do v8 😅